statefun-flink/statefun-flink-core/src/main/java/org/apache/flink/statefun/flink/core/reqreply/RequestReplyFunction.java

```diff
@@ -241,6 +242,16 @@ private void handleOutgoingDelayedMessages(Context context, InvocationResponse i
     }
   }
 
+  private void handleDelayedCancellations(InternalContext context, InvocationResponse result) {
+    for (FromFunction.DelayCancellation delayCancellation :
+        result.getOutgoingDelayCancellationsList()) {
+      String token = delayCancellation.getCancellationToken();
+      if (!token.isEmpty()) {
+        context.cancelDelayedMessage(delayCancellation.getCancellationToken());
```

Comment:
Assume the following scenario:
Within the same `FromFunction` response from a remote function, there is a delayed message with cancellation token `foobar`. At the same time, there is also a cancellation token for `foobar`.

I'm wondering if we want to be diligent in this scenario and eagerly "cancel" out the delayed message, without going through a roundtrip of adding it to the buffer / index state, just to delete it almost imediately.

Reply:
I think this might even be a common case.

Say, for the same invocation batch with a given key, an earlier message triggers a delayed message that is cancellable. Maybe as a timer to purge state etc. in the absence of a follow up event.

However, within that same batch, the follow up event is processed and users immediately wants to cancel the delayed message. We can maybe optimize this scenario?